x-pack/legacy/plugins/siem/public/containers/detection_engine/signals/use_signal_index.test.tsx

```diff
@@ -0,0 +1,127 @@
+      });
+    });
+  });
+
+  test('fetch signals info', async () => {
+    await act(async () => {
+      const { result, waitForNextUpdate } = renderHook<void, ReturnSignalIndex>(() =>
+        useSignalIndex()
+      );
+      await waitForNextUpdate();
+      await waitForNextUpdate();
+      expect(result.current).toEqual({
+        createDeSignalIndex: result.current.createDeSignalIndex,
+        loading: false,
+        signalIndexExists: true,
+        signalIndexName: 'mock-signal-index',
+      });
+    });
+  });
+
+  test('make sure that createSignalIndex is giving back the signal info', async () => {
+    await act(async () => {
+      const { result, waitForNextUpdate } = renderHook<void, ReturnSignalIndex>(() =>
+        useSignalIndex()
+      );
+      await waitForNextUpdate();
+      await waitForNextUpdate();
+      if (result.current.createDeSignalIndex != null) {
+        await result.current.createDeSignalIndex();
+      }
+      await waitForNextUpdate();
+      expect(result.current).toEqual({
+        createDeSignalIndex: result.current.createDeSignalIndex,
+        loading: false,
+        signalIndexExists: true,
+        signalIndexName: 'mock-signal-index',
+      });
+    });
+  });
+
+  test('make sure that createSignalIndex have been called when trying to create signal index', async () => {
+    const spyOnCreateSignalIndex = jest.spyOn(api, 'createSignalIndex');
+    await act(async () => {
+      const { result, waitForNextUpdate } = renderHook<void, ReturnSignalIndex>(() =>
+        useSignalIndex()
+      );
+      await waitForNextUpdate();
+      await waitForNextUpdate();
+      if (result.current.createDeSignalIndex != null) {
+        await result.current.createDeSignalIndex();
+      }
+      await waitForNextUpdate();
+      expect(spyOnCreateSignalIndex).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  test('if there is an error during createSignalIndex, we should get back signalIndexExists === false && signalIndexName == null', async () => {
+    const spyOnCreateSignalIndex = jest.spyOn(api, 'createSignalIndex');
+    spyOnCreateSignalIndex.mockImplementation(() => {
+      throw new Error('Something went wrong, let see what happen');
+    });
+    await act(async () => {
+      const { result, waitForNextUpdate } = renderHook<void, ReturnSignalIndex>(() =>
+        useSignalIndex()
+      );
+      await waitForNextUpdate();
+      await waitForNextUpdate();
+      if (result.current.createDeSignalIndex != null) {
+        await result.current.createDeSignalIndex();
+      }
+      expect(result.current).toEqual({
+        createDeSignalIndex: result.current.createDeSignalIndex,
+        loading: false,
+        signalIndexExists: false,
+        signalIndexName: null,
+      });
+    });
+  });
+
+  test('if there is an error when fetching signals info, signalIndexExists === false && signalIndexName == null', async () => {
+    const spyOnGetSignalIndex = jest.spyOn(api, 'getSignalIndex');
```

Comment:
All of `api` should already be mocked here, right? Since @dhurley14 made me aware of https://github.com/facebook/jest/issues/7136#issuecomment-565976599 I'm wary of any uses of `spyOn`, but it should be unnecessary here (except to type it as a jest.Mock).